src/i2s.rs

```diff
@@ -0,0 +1,1835 @@
+        }
+
+        /// Convert to the ESP-IDF SDK `i2s_std_config_t` representation.
+        #[inline(always)]
+        pub(crate) fn as_sdk(&self) -> i2s_std_config_t {
+            i2s_std_config_t {
+                clk_cfg: self.clk_cfg.as_sdk(),
+                slot_cfg: self.slot_cfg.as_sdk(),
+                gpio_cfg: self.gpio_cfg.as_sdk(),
+            }
+        }
+    }
+
+    impl<'a, Bclk, Din, Dout, Mclk, Ws> From<&'a StdChanConfig<Bclk, Din, Dout, Mclk, Ws>>
+        for ChanConfig<'a, Bclk, Din, Dout, Mclk, Ws>
+    where
+        Bclk: Pin + IOPin + 'static,
+        Din: MaybePin + 'static,
+        Dout: MaybePin + 'static,
+        Mclk: MaybePin + 'static,
+        Ws: Pin + IOPin + 'static,
+    {
+        #[inline(always)]
+        fn from(cfg: &'a StdChanConfig<Bclk, Din, Dout, Mclk, Ws>) -> Self {
+            Self::Std(cfg)
+        }
+    }
+
+    /// Standard mode channel clock configuration.
+    #[derive(Clone)]
+    pub struct StdClkConfig {
+        /// I2S sample rate.
+        pub sample_rate_hz: u32,
+
+        /// Clock source.
+        pub clk_src: ClockSource,
+
+        /// The multiple of MCLK to the sample rate.
+        pub mclk_multiple: MclkMultiple,
+    }
+
+    impl StdClkConfig {
+        /// Create a standard clock configuration with the specified rate in Hz.
+        ///
+        /// # Note
+        /// Set the mclk_multiple to [MclkMultiple::M384] when using 24-bit data width. Otherwise, the sample rate
+        /// might be imprecise since the BCLK division is not an integer.
+        #[inline(always)]
+        pub fn from_sample_rate_hz(rate: u32) -> Self {
+            Self {
+                sample_rate_hz: rate,
+                clk_src: ClockSource::default(),
+                mclk_multiple: MclkMultiple::M256,
+            }
+        }
+
+        /// Convert to the ESP-IDF SDK `i2s_std_clk_config_t` representation.
+        #[inline(always)]
+        pub(crate) fn as_sdk(&self) -> i2s_std_clk_config_t {
+            i2s_std_clk_config_t {
+                sample_rate_hz: self.sample_rate_hz,
+                clk_src: self.clk_src.as_sdk(),
+                mclk_multiple: self.mclk_multiple.as_sdk(),
+            }
+        }
+    }
+
+    /// Standard mode GPIO (general purpose input/output) configuration.
+    pub struct StdGpioConfig<Bclk, Din, Dout, Mclk, Ws>
+    where
+        Bclk: Pin + IOPin + 'static,
+        Din: MaybePin + 'static,
+        Dout: MaybePin + 'static,
+        Mclk: MaybePin + 'static,
+        Ws: Pin + IOPin + 'static,
+    {
+        /// MCLK (master/controller clock) pin. Optional output.
+        mclk: Mclk,
+
+        /// BCLK (bit clock) pin. Input in target/slave mode, output in controller/master mode.
+        bclk: PeripheralRef<'static, Bclk>,
+
+        /// WS (word select) pin. Input in target/slave mode, output in controller/master mode.
+        ws: PeripheralRef<'static, Ws>,
+
+        /// Data output pin.
+        data_out: Dout,
```

Comment:
Is this extra MaybePin here really necessary? Isn't it possible to use an Option PeripheralRef Output and store it as some OutputPin. Again can get rid of the generics